src/app/shared/components/upsert-address/upsert-address.component.ts

```diff
@@ -128,4 +158,7 @@ export class UpsertAddressComponent implements OnInit {
     this.addressForm.controls['latitude'].setValue(data.coords.lat);
     this.addressForm.controls['longitude'].setValue(data.coords.lng);
   }
+
+  addAddress() {
```

Comment:
There is submitAddress function to handle upsert functionality, so no need to addAddress function